media-proxy/src/mesh/conn_rdma.cc

```diff
@@ -0,0 +1,388 @@
+#include <queue>
+#include "conn_rdma.h"
+
+namespace mesh {
+
+namespace connection {
+
+Rdma::Rdma() : ep_ctx(nullptr), init(false), trx_sz(0), mDevHandle(nullptr)
+{
+    std::memset(&ep_cfg, 0, sizeof(ep_cfg));
+}
+
+Rdma::~Rdma() { shutdown_rdma(context::Background()); }
+
+std::string Rdma::kind_to_string(Kind kind)
+{
+    switch (kind) {
+    case Kind::undefined:
+        return "undefined";
+    case Kind::transmitter:
+        return "transmitter";
+    case Kind::receiver:
+        return "receiver";
+    default:
+        return "unknown";
+    }
+}
+
+void Rdma::notify_cq_event()
+{
+    std::lock_guard<std::mutex> lock(cq_mutex);
+    event_ready = true;
+    cq_cv.notify_one();
+}
+
+void Rdma::init_buf_available() { buf_available = false; }
+
+void Rdma::notify_buf_available()
+{
+    buf_available.store(true, std::memory_order_release);
+    buf_available.notify_one();
+}
+
+void Rdma::wait_buf_available()
+{
+    buf_available.wait(false, std::memory_order_acquire);
+    buf_available = false;
+}
+
+// Add to queue
+Result Rdma::add_to_queue(void *element)
+{
+    if (!element) {
+        return Result::error_bad_argument;
+    }
+
+    {
+        std::lock_guard<std::mutex> lock(queue_mutex);
+        buffer_queue.push(element);
+    }
+    notify_buf_available(); // Notify availability
+
+    return Result::success;
+}
+
+// Consume from queue
+Result Rdma::consume_from_queue(context::Context& ctx, void **element)
+{
+    std::unique_lock<std::mutex> lock(queue_mutex);
+
+    if (ctx.cancelled()) {
+        *element = nullptr; // Ensure element remains nullptr
+        return Result::error_operation_cancelled;
+    }
+
+    if (buffer_queue.empty()) {
+        return Result::error_no_buffer;
+    }
+
+    *element = buffer_queue.front();
+    buffer_queue.pop();
+
+    return Result::success;
+}
+
+/**
+ * @brief Initializes the RDMA buffer queue with a specified number of pre-allocated elements.
+ * 
+ * Allocates a single contiguous memory block, aligns buffer sizes to the page size, and divides 
+ * the block into individual buffers. Pushes these buffers into the queue for future use.
+ * Ensures proper cleanup on failure or reinitialization attempts.
+ * 
+ * @param capacity The number of buffers to allocate within continuous virtual memory.
+ * @param trx_sz The size of each buffer (aligned to the page size).
+ * @return Result::success on successful initialization, or an error result if arguments are invalid 
+ *         or memory allocation fails.
+ */
+Result Rdma::init_queue_with_elements(size_t capacity, size_t trx_sz)
+{
+    if (trx_sz == 0 || trx_sz > MAX_BUFFER_SIZE) {
+        log::error("Invalid transfer size provided for buffer allocation")("trx_sz", trx_sz);
+        return Result::error_bad_argument;
+    }
+
+    std::lock_guard<std::mutex> lock(queue_mutex);
+
+    // Check if already initialized
+    if (!buffer_queue.empty()) {
+        log::error("Buffer queue already initialized");
```

Comment:
Make the log message more descriptive. We have numerous components in Media Proxy, so the logs must bring more details.
```suggestion
        log::error("RDMA buffer queue already initialized");
```